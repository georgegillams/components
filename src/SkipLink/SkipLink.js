/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016-2020 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import PropTypes from 'prop-types';
import React from 'react';

import { cssModules } from '../helpers/cssModules';

import STYLES from './skip-link.scss';

const getClassName = cssModules(STYLES);

const SkipLink = props => {
  const { label, href, className, ...rest } = props;

  return (
    <a href={href} className={getClassName('skip-link', className)} {...rest}>
      {label}
    </a>
  );
};

SkipLink.propTypes = {
  label: PropTypes.string.isRequired,
  href: PropTypes.string.isRequired,
  className: PropTypes.string,
};

SkipLink.defaultProps = {
  className: null,
};

export default SkipLink;
