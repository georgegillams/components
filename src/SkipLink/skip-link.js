import PropTypes from 'prop-types';
import React from 'react';
import { Link, StyledSkipLink, Wrapper } from './skip-link.styles';

const SkipLink = (props) => {
  const { label, href, className, ...rest } = props;

  return (
    <Wrapper {...rest}>
      <Link href={href}>{label}</Link>
    </Wrapper>
  );
};

SkipLink.propTypes = {
  label: PropTypes.string.isRequired,
  href: PropTypes.string.isRequired,
};

SkipLink.defaultProps = {};

export default SkipLink;
