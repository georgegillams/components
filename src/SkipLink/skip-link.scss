/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016-2020 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

@import '../Tokens/common';

.skip-link {
  display: inline-block;
  padding: $spacing-sm * 2;
  transition: none;
  border-radius: $bpk-border-radius-sm;
  background-color: $bpk-color-panjin;
  color: white;
  font-weight: bold;
  text-decoration: none;
  opacity: 1;
  overflow: hidden;

  &--hidden {
    transition: opacity $duration-sm ease-in-out;
    opacity: 0;
    pointer-events: none;
  }
}
