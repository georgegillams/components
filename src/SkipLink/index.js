import SkipLink from './SkipLink';

export default SkipLink;
export { SkipLink };
