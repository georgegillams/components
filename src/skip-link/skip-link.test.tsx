import React from 'react';
import renderer from 'react-test-renderer';

import SkipLink from './index';

describe('SkipLink', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(<SkipLink href="#main" label="Skip to main content" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should support arbitrary props', () => {
    const tree = renderer
      .create(
        // @ts-ignore
        <SkipLink href="#main" label="Skip to main content" testID="123" />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
