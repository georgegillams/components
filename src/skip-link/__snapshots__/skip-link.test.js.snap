// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkipLink should render correctly 1`] = `
<div
  className="sc-bcXHqe fzwuJr"
>
  <a
    className="sc-gswNZR irEuey"
    href="#main"
  >
    Skip to main content
  </a>
</div>
`;

exports[`SkipLink should support arbitrary props 1`] = `
<div
  className="sc-bcXHqe fzwuJr"
>
  <a
    className="sc-gswNZR irEuey"
    href="#main"
  >
    Skip to main content
  </a>
</div>
`;

exports[`SkipLink should support custom class names 1`] = `
<div
  className="sc-bcXHqe fzwuJr custom-classname"
>
  <a
    className="sc-gswNZR irEuey"
    href="#main"
  >
    Skip to main content
  </a>
</div>
`;
